logstash-core/lib/logstash/config/config_ast.rb

```diff
@@ -6,31 +6,42 @@
 
 module LogStash; module Config; module AST
   PROCESS_ESCAPE_SEQUENCES = :process_escape_sequences
+  MUTEX = Mutex.new
 
   def self.deferred_conditionals=(val)
+    warn("lock not held (at #{caller.first})") unless MUTEX.owned?
```

Comment:
I am wondering about the usage of `warn` here; this only writes the given message to STDOUT, not sure how useful that is? is there a condition where these methods should be called outside the synchronize block? if not why not just `raise` if this is an unexpected condition?